command-expansion-server/src/app.ts

```diff
@@ -109,52 +143,138 @@ app.post('/put-expansion-set', async (req, res) => {
+    if (expansion === undefined) {
+      return {
+        activityInstance,
+        commands: null,
+        errors: null,
+      };
+    }
+    const activityTypes = generateTypescriptForGraphQLActivitySchema(activitySchema);
+    return await piscina.run({
+      expansionLogic: expansion.expansionLogic,
+      activityInstance,
+      commandTypes,
+      activityTypes,
+    }) as ReturnType<typeof executeExpansion>;
+  }));
 
-app.post('/expand-all-activity-instances/:simulationId(\\d+)/:expansionSetId(\\d+)', async (req, res) => {
-  res.status(501).send('POST /expand-all-activity-instances: Not implemented');
+  const rejectedExpansionResults = settledExpansionResults.filter(isRejected).map(p => p.reason);
+
+  for (const expansionResult of rejectedExpansionResults) {
+    logger.error(expansionResult.reason);
+  }
+  if (rejectedExpansionResults.length > 0) {
+    throw new Error(rejectedExpansionResults.map(rejectedExpansionResult => rejectedExpansionResult.reason).join('\n'));
+  }
+
+  const expandedActivityInstances = settledExpansionResults.filter(isResolved).map(p => ({
+    id: p.value.activityInstance.id,
+    commands: p.value.commands,
+    errors: p.value.errors
+  }));
+
+  // Store expansion run  and activity instance commands in DB
+  const { rows } = await db.query(`
+    WITH expansion_run_id AS (
+      INSERT INTO expansion_run (simulation_dataset_id, expansion_set_id)
+        VALUES ($1, $2)
+        RETURNING id
+    )
+    INSERT
+    INTO activity_instance_commands (expansion_run_id,
+                                     activity_instance_id,
+                                     commands,
+                                     errors)
+    SELECT *
+    FROM unnest(
+        array_fill((SELECT id FROM expansion_run_id), ARRAY [array_length($3::int[], 1)]),
+        $3::int[],
+        $4::jsonb[],
+        $5::jsonb[]
+      )
+    RETURNING (SELECT id FROM expansion_run_id);
+    `, [
+    simulationDatasetId,
+    expansionSetId,
+    expandedActivityInstances.map(result => result.id),
+    expandedActivityInstances.map(result => result.commands !== null ? JSON.stringify(result.commands) : null),
+    expandedActivityInstances.map(result => JSON.stringify(result.errors)),
+  ]);
+
+  if (rows.length < 1) {
+    throw new Error(`POST /expand-all-activity-instances: No expansion run was inserted in the database`);
+  }
+  const id = rows[0].id;
+  logger.info(`POST /expand-all-activity-instances: Inserted expansion run to the database: id=${id}`);
+
+  res.status(200).json({
+    id,
+    expandedActivityInstances,
+  });
   return;
 });
 
+// General error handler
 app.use((err: any, req: Request, res: Response, _next: NextFunction) => {
   logger.error(err);
-  res.status(err.status ?? err.statusCode ?? 500).send(err.message);
+  res.status(err.status ?? err.statusCode ?? 500).send({message: err.message});
+  return _next();
```

Comment:
Can we remove the leading underscore in `_next` so that it isn't confused as being an unused parameter?